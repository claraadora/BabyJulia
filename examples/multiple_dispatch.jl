abstract type Human end 
struct BabyHuman <: Human
    name::String 
    pottyTrainingLevel::Int64
end 
struct AdultHuman <: Human
    name::String
    job::String
end 

abstract type Emotion end 
struct Happy <: Emotion
    name::String
end 
struct Sad <: Emotion
    name::String
end 

# Construct 
julia = BabyHuman("Julia", 1)
profIlya = AdultHuman("Prof Ilya", "Professor")

happy = Happy("happy")
sad = Sad("sad")

function setMood(human::Human, emotion::Emotion) 
    return human.name * ": " * "* mood set to " * emotion.name * " *"
end

function setMood(baby::BabyHuman, emotion::Sad) 
    return baby.name * ": " * "hue hue hue hue....."
end

function setMood(baby::AdultHuman, emotion::Sad) 
    return baby.name * ": " * "I'm feeling blue..."
end

println(setMood(julia, happy))
println(setMood(profIlya, happy))

println(setMood(julia, sad))
setMood(profIlya, sad)