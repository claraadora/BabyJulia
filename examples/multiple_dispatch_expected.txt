"Prof Ilya: Im feeling blue..."